service/worker/scheduler/workflow.go

```diff
@@ -60,6 +60,8 @@ const (
 	InitialVersion SchedulerWorkflowVersion = iota
 	// skip over entire time range if paused and batch and cache getNextTime queries
 	BatchAndCacheTimeQueries
+	// Delete idle schedule after a timer fires
+	DeleteIdleSchedule
```

Comment:
you could just use `tweakables.RetentionTime > 0` in this case, since it's a new field